src/PositionUtilities.php

```diff
@@ -26,26 +26,16 @@ public static function getRangeFromPosition($pos, $length, $text) {
      * @return LineCharacterPosition
      */
     public static function getLineCharacterPositionFromPosition($pos, $text) : LineCharacterPosition {
-        if ($pos >= \strlen($text)) {
-            $pos = \strlen($text);
+        $length = \strlen($text);
+        if ($pos >= $length) {
+            $pos = $length;
         } elseif ($pos < 0) {
             $pos = 0;
         }
 
-        $newlinePositions = [];
-        $newlinePos = -1;
-        while ($newlinePos = \strpos($text, "\n", $newlinePos + 1)) {
-            if ($newlinePos < $pos) {
-                $newlinePositions[] = $newlinePos;
-                continue;
-            }
-            break;
-        }
-
-        $lastNewline = \count($newlinePositions) - 1;
-        $char = $pos - ($lastNewline >= 0 ? $newlinePositions[$lastNewline] + 1 : 0);
-        $line = \count($newlinePositions);
-
+        $lastNewlinePos = \strrpos($text, "\n", -($length - $pos));
+        $char = $pos - ($lastNewlinePos === false ? 0 : $lastNewlinePos + 1);
+        $line = $pos > 0 ? \substr_count($text, "\n", 0, $pos) : 0;
```

Comment:
Looks like an implementation bug. The RFC says it has the "same behavior as substr", which does the right thing for length=0. As 7.1 is early in the release cycle and this is new functionality (in PHP 7.0 length=0 was not allowed), we can fix this without problems. Can you please file a bug on bugs.php.net?